Fixes #1108 by only throwing exception when registering region with c…

scottpurdy opened the pull request:
…lass name that is already registered if the new region is from a different module than the original.

fixes #1108 

@mrcslws please review

rhyolight commented:
@scottpurdy A test called `testFailOnRegisterDuplicateRegion` is failing on Windows. I wonder why it is not failing on other OSes?

scottpurdy commented:
@rhyolight looks like cpp_region_test was run in most places but py_region_test was not. I added it to a couple more places (travis, bamboo os x) and fixed the test.

scottpurdy commented:
@mrcslws - I agree. It's just annoying because it has to be a region with the same name but in a different module. But probably better to add that now.

vitaly-krugl commented:
@mrcslws, concerning 
> But I don't feel strongly about that. I've written my fair share of non-unit-tested exceptions.

Is this because we have reasonable confidence that throwing an exception would "just work" in a strongly-type language like C++ versus the likes of Python?
